harness/determined/launch/horovod.py

```diff
@@ -179,6 +188,18 @@ def main(hvd_args: List[str], script: List[str], autohorovod: bool) -> int:
 
     os.environ["USE_HOROVOD"] = "1"
 
+    # We now have environment images with built-in OpenMPI.   When invoked the
+    # SLURM_JOBID variable triggers integration with SLURM, however, we are
+    # running in a singularity container and SLURM may or may not have
+    # compatible configuration enabled.  We therefore clear the SLURM_JOBID variable
+    # before invoking mpi so that mpirun will honor the args passed via horvod
+    # run to it describing the hosts and process topology, otherwise mpi ends
+    # up wanting to launch all -np# processes on the local causing an oversubscription
+    # error ("There are not enough slots available in the system").
+    slurm_vars_to_clear = ["SLURM_JOBID"]
+    for k, _ in os.environ.items():
+        if k in slurm_vars_to_clear:
+            os.environ.pop(k)
```

Comment:
Maybe just a plain:
```python
os.environ.pop("SLURM_JOBID", None)
```
and we can make a loop out of it when we need it to be a loop.